### DISCUSSION TITLE
Action secrets and pull requests from non-contributors on OSS project

### DISCUSSION THREAD

**Discussion by coaxial** [community/community#63116] [Actions] [answered]
### Select Topic Area

Question

### Body

Our open source project has all the secrets it needs defined in the settings, and this works normally.

When someone who isn't a project member contributes a PR, the actions fail because the secrets aren't accessible anymore after authorizing the workflows to run.

See https://github.com/maid/maid/actions/runs/5775015686/job/15704394069?pr=323 (❌ fails, from non-member) vs https://github.com/maid/maid/actions/runs/4958403837/job/13432992724 (✅ passes, from project member)

Is there a way to have the action and the secrets also run on PRs from non project members?

**Comment by mvkaran** [comment, marked as answer]
Hi @coaxial 👋🏼 

It looks like the PR that you shared where a workflow run was failing, originated from a fork. Actions secrets are not passed on to workflows originating from a fork [by design](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories) due to security considerations.

An alternative is the [`pull_request_target` trigger](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) that provides access to the secrets as well. However, please be extremely careful and sure that you want the secrets to be shared with a PR from a fork, since the PR author could misuse it. 

Another alternative could be to re-architect your workflow in a way that the workflow which triggers from a PR uploads some artifacts from it, and further triggers another workflow that could run in the context of your repository which can then perform any tasks on those artifacts using the secrets.

**Reply by coaxial** [reply]
Thank you, will investigate the solutions you shared :)
